src/typing/typer.ml

```diff
@@ -718,7 +718,7 @@ and type_vars ctx vl p =
 	List.iter (fun (v,_) ->
 		delay_if_mono ctx PTypeField v.v_type (fun() ->
 			if ExtType.is_void (follow v.v_type) then
-				raise_typing_error "Variables of type Void are not allowed" v.v_pos
+				raise_typing_error "Variables of type Void are not allowed" (if v.v_pos == null_pos then p else v.v_pos)
```

Comment:
Sounds like parser does something with TVars to avoid that situation, but it doesn't happen for macros. Maybe the actual fix would be to do the same for macro generated TVars?

Reply:
nvm found a proper fix...